coffeecard/CoffeeCard.WebApi/Services/AccountService.cs

```diff
@@ -40,9 +45,24 @@ public string Login(string username, string password, string version)
 
             ValidateVersion(version);
 
+            var timeout = _identitySettings.TimeOut;
+            var maxAttempts = _identitySettings.MaximumLoginAttempts;
+
             var user = _context.Users.FirstOrDefault(x => x.Email == username && x.IsVerified);
             if (user != null)
             {
+                var mapEntry = _loginLimiter.UpdateAndGetLoginAttemptCount(user.Email);
+
+                var loginAttemptsMade = mapEntry.Item2;
+                var timeSinceFirstFailedLogin = DateTime.UtcNow.Subtract(mapEntry.Item1);
+                if (loginAttemptsMade == maxAttempts && timeSinceFirstFailedLogin < timeout)
+                {
+                    Log.Warning("Login attempts exceeding maximum allowed for e-mail = {username} from IP = {ipadress} ", username,
+                        _httpContextAccessor.HttpContext.Connection.RemoteIpAddress);
+                    throw new ApiException($"Amount of failed login attempts exceeds the allowed amount, please wait for {timeout.TotalMinutes} minutes",
```

Comment:
Consider throwing a more description exception instead of the generic ApiException

Reply:
http 429 is for "too many connections" so thought it would be appropriate, any compelling reason not to use it?